### PR TITLE
`ultralytics 8.3.198` Improve `Tuner` with BLX-α gene crossover

### PR DESCRIPTION
**Pull request by glenn-jocher**
image

image

### Reproduce
```python
from ultralytics import YOLO

model = YOLO("yolo11s.yaml")

model.tune(
    device=0,
    data="coco128.yaml",
    optimizer="AdamW",
    epochs=100,
    batch=8,
    compile=False,
    plots=False,
    val=False,
    save=False,
    workers=16,
    project="tune-yolo11s-scratch-coco128-100e",
    iterations=1000,
)
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Smarter, more stable hyperparameter tuning with improved fitness scoring, a new augmentation control, and better resource cleanup — released in ultralytics 8.3.198. 🚀

### 📊 Key Changes
- Version bump to 8.3.198.
- Tuner overhaul:
  - Added BLX-α crossover with weighted parent selection for better exploration. 🔀
  - Mutations now use log-normal steps with adaptive sigma decaying from 0.2 → 0.1 over 300 iters.
  - Introduced new tunable hyperparameter: `close_mosaic` (0–10 epochs, cast to int). 🧩
  - Adjusted bounds: `cls` lower bound from 0.2 → 0.1.
  - More robust parent selection from MongoDB/CSV; safer weights handling.
  - Fixed best-iteration indexing and directory cleanup logic.
  - Added GPU/CPU memory cleanup after each iteration (`gc.collect()` + `torch.cuda.empty_cache()`).
- Fitness metric update:
  - Fitness now uses only mAP@0.5:0.95 (weights: [P, R, mAP@0.5, mAP@0.5:0.95] = [0, 0, 0, 1]). 🎯
- Docs: Added @UltralyticsAbi to docs author map.

### 🎯 Purpose & Impact
- Better tuning quality: BLX-α crossover and refined mutations improve search diversity and convergence, leading to stronger models with fewer iterations. 📈
- More relevant scoring: Focusing on mAP@0.5:0.95 aligns tuning with widely used benchmarks; results may shift compared to previous versions. ✅
- New control for augmentation: `close_mosaic` helps control when mosaic augmentation is disabled, potentially improving late-epoch fine-tuning. 🧪
- Stability and efficiency: Memory cleanup reduces VRAM fragmentation and out-of-memory issues during long tuning runs. 🧠
- Cleaner outputs: Corrected best-iteration tracking and safer directory cleanup reduce disk usage and confusion. 🧹

Tip: If you’re scripting the tuner, you can now include and expect an integer `close_mosaic` in hyperparameters:
```python
from ultralytics.engine.tuner import Tuner

tuner = Tuner()
tuner(iterations=50)  # will tune including 'close_mosaic' as an int
```

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help streamline review. An Ultralytics engineer will also assist you shortly. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any relevant issues using the Ultralytics Issues page: [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics Continuous Integration checks are passing. See the CI guide here: [Ultralytics CI Guide](https://docs.ultralytics.com/help/CI/). If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant documentation for any new or modified features, including tuner behavior, new args like `close_mosaic`, evolutionary strategy notes, and fitness changes: [Ultralytics Documentation](https://docs.ultralytics.com/).
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes (e.g., BLX-α crossover selection, mutation bounds and types, CSV/Mongo fallback, fitness computation). Confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our Contributor License Agreement if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message: [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/).
- ✅ **Minimize Changes**: Limit your changes to the minimum necessary for your feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

Additional notes for this PR:
- 🧬 Thanks for adding BLX-α crossover and tuning updates. Please ensure the PR description clearly states any API-impacting changes (even if backwards-friendly) and where users might see different results due to the fitness update.
- 🧹 For the GPU memory cleanup additions, consider confirming behavior on both CUDA-available and CPU-only environments (e.g., safe no-op behavior).
- 📈 With fitness now based purely on mAP@0.5:0.95, please confirm any docs or examples referencing prior weighted fitness are updated to avoid confusion for users migrating tuning runs.
- 🧪 Where possible, include lightweight tests or usage snippets validating that parameter constraints and casting (e.g., `close_mosaic` int handling) behave as intended.

For more guidance, please refer to our Contributing Guide: [Ultralytics Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Comment by UltralyticsAssistant** [comment]
Huge congratulations on the merge, @glenn-jocher! “Simplicity is the ultimate sophistication.” — Leonardo da Vinci. Your focus on a cleaner fitness signal (mAP@0.5:0.95), smarter exploration (BLX-α + adaptive mutations), the new `close_mosaic` control, and vigilant memory cleanup makes tuning both sharper and more stable—exactly the kind of refinement that moves YOLO forward with confidence and clarity.

Thank you for leading with precision and care—this raises the bar for everyone. Details: [Smarter, more stable hyperparameter tuning in 8.3.198](https://github.com/ultralytics/ultralytics/pull/22038)

Upgrade with: `pip install -U ultralytics`
